shapely/geometry/base.py

```diff
@@ -435,20 +435,29 @@ def buffer(
         4.0
 
         """
+        quadsegs = kwargs.pop("quadsegs", None)
         if quadsegs is not None:
             warn(
-                "The `quadsegs` argument is deprecated. Use `resolution`.",
-                DeprecationWarning,
+                "The `quadsegs` argument is deprecated. Use `quad_segs` instead.",
+                FutureWarning,
             )
-            resolution = quadsegs
+            quad_segs = quadsegs
+
+        # TODO deprecate `resolution` keyword in the future as well
```

Comment:
It seems like the time to go ahead and deprecate this under shapely 2.0 instead of doing so later.

Reply:
> It seems like the time to go ahead and deprecate this under shapely 2.0 instead of doing so later.

I don't feel very strong about this, but so I gave some rationale for not yet deprecating it in the top post: keeping it working (for now) without warning makes it easier to write code that is compatible with both <=1.8 and >=2.0. If we start deprecating it, any other package that uses this and wants to initially support both 1.8 and 2.0 needs to do an if/else based on the shapely version to avoid showing the warning to users.

(one might think that there are already other cases that would need such version checks with the many changes in shapely 2.0, but for example all breaking changes for which we warned in 1.8 can all be solved by using already existing APIs)